applications/ConvectionDiffusionApplication/custom_elements/symbolic_d_convection_diffusion_explicit.cpp

```diff
@@ -0,0 +1,1019 @@
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicDConvectionDiffusionExplicit<TDim,TNumNodes>::Initialize(
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    BaseType::Initialize(rCurrentProcessInfo);
+    mUnknownSubScale = ZeroVector(TNumNodes);
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicDConvectionDiffusionExplicit<TDim,TNumNodes>::FinalizeSolutionStep(
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    BaseType::FinalizeSolutionStep(rCurrentProcessInfo);
+    // Element data
+    ElementData rData;
+    this->InitializeEulerianElement(rData,rCurrentProcessInfo);
+    // Reading integration points and local gradients
+    const auto& r_geometry = this->GetGeometry();
+    const auto& integration_points = r_geometry.IntegrationPoints(this->GetIntegrationMethod());
+
+    // Iterate over integration points to update subscales on the gauss point
+    for (unsigned int g = 0; g < integration_points.size(); g++) {
+        // Caluclate N on the gauss point "g"
+        rData.N = row(rData.N_gausspoint,g);
+        // Compute tau
+        this->CalculateTau(rData);
+        // Retrieve unknown belonging to subgrid scale space on gauss integration point g
+        rData.unknown_subscale = mUnknownSubScale(g);
+        // Update unknown belonging to subgrid scale space on gauss integration point g
+        this->UpdateUnknownSubgridScaleGaussPoint(rData,g);
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicDConvectionDiffusionExplicit<TDim,TNumNodes>::Calculate(
+    const Variable<double>& rVariable,
+    double& Output,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    if (rVariable == SCALAR_PROJECTION) {
+        auto& r_geometry = this->GetGeometry();
+        const unsigned int local_size = r_geometry.size();
+        BoundedVector<double, TNumNodes> rhs_oss;
+        this->CalculateOrthogonalSubgridScaleRHSInternal(rhs_oss,rCurrentProcessInfo);
+        for (unsigned int i_node = 0; i_node < local_size; i_node++) {
+            #pragma omp atomic
+            r_geometry[i_node].GetValue(rVariable) += rhs_oss[i_node];
+        }
+    }
+    else {
+        KRATOS_ERROR << "Variable not implemented to compute OSS projection. Use SCALAR_PROJECTION instead." << std::endl;
```

Comment:
```suggestion
        BaseType::Calculate(rVariable, Output, rCurrentProcessInfo);
```
Actually I think that the implementation of the `SCALAR_PROJECTION` variable calculation is the same in the base class.

Reply:
Done. I did in the template, thus do not apply directly your suggestion.